src/core/ucc_context.c

```diff
@@ -415,9 +506,16 @@ ucc_status_t ucc_context_create(ucc_lib_h lib,
         ucc_error("failed to init progress queue for context %p", ctx);
         goto error_ctx_create;
     }
-    ctx->id.host_id = ucc_local_proc.host_id;
-    ctx->id.pid     = getpid();
-    ctx->id.seq_num = ucc_context_seq_num++;
+    ctx->id.pi      = ucc_local_proc;
+    ctx->id.seq_num = ucc_atomic_fadd32(&ucc_context_seq_num, 1);
```

Comment:
Is atomic really needed? I thought it’s not possible to call ucc control flow api from different threads

Reply:
yes, it turned out to be needed, if a thread  runs a separate ucc context and thus behaves as a "rank" - e.g. this happens in gtest

Reply:
what is the use case of this ID?

Reply:
it is used to uniquely identify a participant of ucc context (a process) - for example, it is used in tl/ucp as a key to the hash table of the UCP eps.